openedx/core/djangoapps/lang_pref/middleware.py

```diff
@@ -39,9 +39,16 @@ def process_request(self, request):
             preferred_language = request.META.get('HTTP_ACCEPT_LANGUAGE', '')
             lang_headers = [seq[0] for seq in parse_accept_lang_header(preferred_language)]
 
+            prefixes = [prefix.split("-")[0] for prefix in system_released_languages]
```

Comment:
@nasthagiri I absolutely agree. We could probably just dupe some of the effort for timezones and change the new context processor. (as it handles lang settings now). I'd also like to crawl back through and eliminate some of the SQL queries.